src/flags.rs

```diff
@@ -0,0 +1,478 @@
+    force_frame_pointers: Option<bool>,
+    link_dead_code: Option<bool>,
+    no_redzone: Option<bool>,
+    soft_float: Option<bool>,
+}
+
+impl RustcCodegenFlags {
+    // Parse flags obtained from CARGO_ENCODED_RUSTFLAGS
+    pub fn parse(rustflags_env: &str) -> RustcCodegenFlags {
+        fn is_flag_prefix(flag: &str) -> bool {
+            match flag {
+                "-Z" | "-C" | "--codegen" | "-L" | "-l" | "-o" => true,
+                "-W" | "--warn" | "-A" | "--allow" | "-D" | "--deny" | "-F" | "--forbid" => true,
+                _ => false,
+            }
+        }
+
+        fn handle_flag_prefix<'a>(prev: &'a str, curr: &'a str) -> Cow<'a, str> {
+            match prev {
+                "--codegen" | "-C" => Cow::from(format!("-C{}", curr)),
+                // Handle flags passed like --codegen=code-model=small
+                _ if curr.starts_with("--codegen=") => Cow::from(format!("-C{}", &curr[10..])),
+                "-Z" => Cow::from(format!("-Z{}", curr)),
+                "-L" | "-l" | "-o" => Cow::from(format!("{}{}", prev, curr)),
+                // Handle lint flags
+                "-W" | "--warn" => Cow::from(format!("-W{}", curr)),
+                "-A" | "--allow" => Cow::from(format!("-A{}", curr)),
+                "-D" | "--deny" => Cow::from(format!("-D{}", curr)),
+                "-F" | "--forbid" => Cow::from(format!("-F{}", curr)),
+                _ => Cow::from(curr),
+            }
+        }
+
+        let mut codegen_flags = RustcCodegenFlags {
+            branch_protection: None,
+            code_model: None,
+            no_vectorize_loops: false,
+            no_vectorize_slp: false,
+            profile_generate: None,
+            profile_use: None,
+            control_flow_guard: None,
+            lto: None,
+            relocation_model: None,
+            embed_bitcode: None,
+            force_frame_pointers: None,
+            link_dead_code: None,
+            no_redzone: None,
+            soft_float: None,
+        };
+
+        let env_flags: Vec<&str> = rustflags_env.split("\u{1f}").collect();
+
+        for i in 0..env_flags.len() {
+            let curr = env_flags[i];
+            let prev = if i > 0 { env_flags[i - 1] } else { "" };
+
+            // Do not process prefixes on their own
+            if !is_flag_prefix(curr) {
+                // Concat flags preceded by a prefix
+                let rustc_flag = handle_flag_prefix(prev, curr);
+                codegen_flags.set_rustc_flag(&rustc_flag);
+            }
+        }
+
+        codegen_flags
+    }
+
+    fn set_rustc_flag(&mut self, flag: &str) {
+        // Convert a textual representation of a bool-like rustc flag argument into an actual bool
+        fn arg_to_bool(arg: &str) -> Option<bool> {
+            match arg {
+                "y" | "yes" | "on" | "true" => Some(true),
+                "n" | "no" | "off" | "false" => Some(false),
+                _ => None,
+            }
+        }
```

Comment:
How about just this then? Not much point making it into two functions since it's only used here to avoid code duplication
```
        fn arg_to_bool(arg: impl AsRef<str>) -> Option<bool> {
            match arg.as_ref() {
                "y" | "yes" | "on" | "true" => Some(true),
                "n" | "no" | "off" | "false" => Some(false),
                _ => None,
            }
        }
```

Reply:
Yes that's fine to me, the reason I create two function is to avoid duplicate code monomorphisation, but it probably won't make much difference in real life so go for the simpler one please.